standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/rules/Mariadb.java

```diff
@@ -18,19 +18,31 @@
 
 package org.apache.hadoop.hive.metastore.dbinstall.rules;
 
+import org.apache.hadoop.hive.metastore.dbinstall.AbstractDatabase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-public class Mariadb extends DatabaseRule {
+public class Mariadb extends AbstractDatabase {
 
   @Override
   public String getDockerImageName() {
     return "mariadb:10.2";
   }
 
   @Override
-  public String[] getDockerAdditionalArgs() {
-    return buildArray("-p", "3306:3306", "-e", "MYSQL_ROOT_PASSWORD=" + getDbRootPassword(), "-d");
+  public String getDockerDatabaseArg() {
+    return "MARIADB_DATABASE=" + getDb();
+  }
+
+  @Override
+  public List<String> getDockerBaseArgs() {
+    return new ArrayList(Arrays.asList("-p", "3309:3306",
```

Comment:
Why are we wrapping into ArrayList? Are we going to modify the list? Keeping things immutable makes the code easier to understand and use.

Reply:
I was wrapping this because further database argumentum can be added in the AbstractDatabase class, so the list can’t be immutable. If we are getting rid of the docker database argument this will be removed.

Reply:
I think that we could create a new list inside `AbstractDatabse.getDockerAdditionalArgs()` if necessary instead of modifying those that we obtain from the getters.